engine/orchestration/src/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java

```diff
@@ -2860,7 +2860,7 @@ public boolean restartNetwork(final Long networkId, final Account callerAccount,
         s_logger.debug("Implementing the network " + network + " elements and resources as a part of network restart without cleanup");
         try {
             implementNetworkElementsAndResources(dest, context, network, offering);
-            setRestartRequired(network, true);
+            setRestartRequired(network, false);
```

Comment:
This fixes an issue where restarting a network without cleanup always put the network's restart required flag to true.

Reply:
this will the opposite? I mean, if I restart with cleanup, will not this cause the flag to always be false? Or, is the flag from the command captured and used somewhere else?

Reply:
@rafaelweingartner the `restart required` flag is a presentational issue and used to communicate to users/admins that a network requires a restart because likely a previous restart attempt failed..

If you restart a network current without the cleanup=true flag, in the API response and UI the network's restart required state will change to true which is wrong. The state/flag should only be true if something went wrong in its last restart. You may read a recent thread here: https://markmail.org/message/bn57hymlvohf54b7